apps/precinct-scanner/src/AppRoot.tsx

```diff
@@ -10,14 +18,193 @@ import { Storage } from './utils/Storage'
 // import ScanSuccessScreen from './screens/ScanSuccessScreen'
 // import ScanWarningScreen from './screens/ScanWarningScreen'
-import UnconfiguredScreen from './screens/UnconfiguredScreen'
+
+import {
+  getStatus as usbDriveGetStatus,
+  doMount,
+  UsbDriveStatus,
+} from './utils/usbstick'
+import { ISO8601Timestamp } from './config/types'
+import * as config from './api/config'
+
+export interface AppStorage {
+  electionDefinition?: ElectionDefinition
+  configuredAt?: ISO8601Timestamp
+}
 
 export interface Props extends RouteComponentProps {
   storage: Storage
 }
 
-const AppRoot: React.FC<Props> = () => {
-  return <UnconfiguredScreen />
+export const electionDefinitionStorageKey = 'electionDefinition'
+export const configuredAtStorageKey = 'configuredAt'
+
+interface HardwareState {
+  hasCardReaderAttached: boolean
+  hasPrinterAttached: boolean
+  usbDriveStatus: UsbDriveStatus
+  // machineConfig: Readonly<MachineConfig>
+}
+
+interface SharedState {
+  electionDefinition: OptionalElectionDefinition
+  isScannerConfigured: boolean
+  isTestMode: boolean
+}
+
+export interface State extends HardwareState, SharedState {}
+
+const initialHardwareState: Readonly<HardwareState> = {
+  hasCardReaderAttached: true,
+  hasPrinterAttached: true,
+  usbDriveStatus: UsbDriveStatus.absent,
+  // machineConfig: { machineId: '0000', codeVersion: 'dev' },
+}
+
+const initialSharedState: Readonly<SharedState> = {
+  electionDefinition: undefined,
+  isScannerConfigured: false,
+  isTestMode: false,
+}
+
+const initialAppState: Readonly<State> = {
+  ...initialHardwareState,
+  ...initialSharedState,
+}
+
+// Sets State.
+type AppAction =
+  | { type: 'unconfigureScanner' }
+  | {
+      type: 'updateElectionDefinition'
+      electionDefinition: OptionalElectionDefinition
+    }
+  | { type: 'updateUsbDriveStatus'; usbDriveStatus: UsbDriveStatus }
+  | {
+      type: 'refreshConfigFromScanner'
+      electionDefinition: OptionalElectionDefinition
+      isTestMode: boolean
+    }
+
+const appReducer = (state: State, action: AppAction): State => {
+  switch (action.type) {
+    case 'updateUsbDriveStatus':
+      return {
+        ...state,
+        usbDriveStatus: action.usbDriveStatus,
+      }
+    case 'updateElectionDefinition':
+      return {
+        ...state,
+        electionDefinition: action.electionDefinition,
+      }
+    case 'refreshConfigFromScanner':
+      return {
+        ...state,
+        electionDefinition: action.electionDefinition,
+        isTestMode: action.isTestMode,
+        isScannerConfigured: true,
+      }
+    case 'unconfigureScanner':
+      return {
+        ...state,
+        isScannerConfigured: false,
+      }
+  }
+}
+
+const AppRoot: React.FC<Props> = (/* { storage } */) => {
+  const [appState, dispatchAppState] = useReducer(appReducer, initialAppState)
+  const {
+    usbDriveStatus,
+    // machineConfig,
+    electionDefinition,
+    // isTestMode,
```

Comment:
should this one not be commented out, given that you're using this appState field in later code?